src/sonic-host-services/scripts/hostcfgd

```diff
@@ -447,8 +448,21 @@ class HostConfigDaemon:
         syslog.syslog(syslog.LOG_INFO, 'Kdump handler...')
         self.kdumpCfg.kdump_update(key, data, False)
 
+    def wait_till_system_init_done(self):
+
+        # No need to print the output in the log file so using the "--quiet"
+        # flag
+        systemctl_cmd = "sudo systemctl is-system-running --wait --quiet"
+        subprocess.call(systemctl_cmd, shell=True)
+
+
     def start(self):
 
+        syslog.syslog(syslog.LOG_INFO,
+                      "Waiting for systemctl to finish initialization")
+        self.wait_till_system_init_done()
+        syslog.syslog(syslog.LOG_INFO, "starting hostcfgd...")
```

Comment:
can we only delay the update_all_feature_states()?

Reply:
Changed in latest commit